Add wilight switch (#22547)
home-assistant/home-assistant.github.io: source/_integrations/wilight.markdown

ha_category:
  - Cover
  - Fan
  - Irrigation
  - Light
  - Switch
ha_release: 0.115
ha_config_flow: true

> ha_iot_class: Local Push

Reviewer: The integration manifest says it's Local Polling.

Author: Fixed!
Thanks.

There is currently support for the following device types within Home Assistant:

- Cover (WiLight model C-103).
- [Fan](#fan) (WiLight model V-104).
- Light (WiLight model I-100, I-102, I-107, I-110 and I-112).
- [Irrigation](#irrigation) / Switch (WiLight model R-105).

{% include integrations/config_flow.md %}

## Fan

> The `wilight` `fan` integration allows you to control your Fans from within Home Assistant.

Reviewer: The integration is wilight. The integration has a fan platform. But I don't think we need to specify that to users. It's enough to say that it allows users to control their wilight fan.

Reviewer: ```suggestion
The `wilight` integration allows you to control your Fans from within Home Assistant.
```

### Fan Services

> There are several services which can be used for automations and control of the `fan`:

Reviewer: ```suggestion
There are several services which can be used for automations and control of the fan:
```

| Service | Description |
| --------- | ----------- |
| `set_percentage` | Calling this service sets the fan speed (`entity_id` and `percentage` are required parameters). Percentage must be in the range 0-100, percentage = 0 turns off the fan. There are three speed levels 33%, 66% and 100%. Adjusting to intermediate levels goes to the next higher level.
| `set_direction` | Calling this service will set the fan direction (`entity_id` and `direction` are required parameters, and direction must be one of the following: forward or reverse). Calling this service will turn the fan on.
| `toggle` | Calling this service will toggle the fan between on and off states (`entity_id` is required).
| `turn_off` | Calling this service will turn the fan off (`entity_id` is required).
| `turn_on` | Calling this service will turn the fan on and set the speed and direction to the last used ones (defaults to high and forward, `entity_id` is required).

## Irrigation

> The `wilight` `switch` integration allows you to control your Irrigation from within Home Assistant.

Reviewer: ```suggestion
The `wilight` integration allows you to control your Irrigation from within Home Assistant. This is exposed via switches in Home Assistant.
```

> There are two switch types in Irrigation: `watering switch` and `pause switch`.

Reviewer: ```suggestion
There are two switch types for Irrigation: `watering switch` and `pause switch`.
```

> `Watering switch` turn on and off the irrigation valve, while the `pause switch` disable / enable the action of `watering switch`.

Reviewer: ```suggestion
A `watering switch` can turn on and off the irrigation valve, while the `pause switch` can disable / enable the action of a `watering switch`.
```

> `Triggers` activate (turn on) the `watering switch` (irrigation valve) at the programmed time, which can be defined on which days of the week it is desired or if only once (today).

Reviewer: ```suggestion
`Triggers` activate (turn on) the `watering switch` (irrigation valve) at the programmed time. The trigger can be set to run on a day of the week or only once (today).
```
